### PR TITLE
Demo scene & DocSections in suggestion database

### PR DESCRIPTION
**Pull request by vitvakatu**
### Pull Request Description

[Task link](https://www.pivotaltracker.com/story/show/184000483)
[Task link](https://www.pivotaltracker.com/story/show/184012363)

This PR adds a new `documentation` demo scene that would be used while developing a new documentation panel.

Also, entries in the Suggestion database now store `DocSection`s provided by the engine, and we can mock them for testing purposes using the new `doc_section!` macro.

No changes in the documentation panel were made.

https://user-images.githubusercontent.com/6566674/207567074-5c5682df-a0e8-435c-b393-6f02b2969614.mp4

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide build` and `./run ide watch`.

### CODE REVIEW

File: app/gui/src/documentation_debug_scene.rs [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+use crate::doc_section;
+use crate::mock_suggestion_database;
+use crate::model::suggestion_database;
+use crate::model::suggestion_database::Entry;
+use crate::model::SuggestionDatabase;
+use ensogl::application::Application;
+use ensogl::data::color;
+use ensogl::display;
+use ensogl::display::navigation::navigator::Navigator;
+use ensogl::display::object::ObjectOps;
```
Comment (wdanilo) [inline app/gui/src/documentation_debug_scene.rs]: not needed

File: app/gui/src/documentation_debug_scene.rs [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+use ide_view::graph_editor::component::visualization::Registry;
+use std::f32::consts::PI;
+
+
+
+// =======================
+// === DatabaseWrapper ===
+// =======================
+
+/// A wrapper around the suggestion database that allows iterate over entries and display their
```
Comment (wdanilo) [inline app/gui/src/documentation_debug_scene.rs]: 1. "wrapper for" is better than "wrapper around"
2. "allows iteration" or "allows to iterate"

Were the sentences checked with gramarly?

File: app/gui/src/documentation_debug_scene.rs [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+    use super::*;
+    shape! {
+        (style: Style) {
+            let background = Rect((BUTTON_SIZE.px(), BUTTON_SIZE.px()));
+            let background = background.corners_radius(10.0.px());
+            let background = background.fill(BUTTON_BACKGROUND_COLOR);
+            let icon = Triangle(10.0, 10.0);
+            let icon = icon.rotate((PI/2.0).radians());
+            let icon = icon.fill(color::Rgba::red());
+
```
Comment (wdanilo) [inline app/gui/src/documentation_debug_scene.rs]: not needed nl

File: app/gui/src/model/suggestion_database/mock.rs [hunk 4]
```diff
@@ -321,6 +322,94 @@ macro_rules! mock_suggestion_database {
...
+/// ```
+/// # use enso_gui::doc_section;
+/// doc_section!(! "Marked as important");
+/// doc_section!(! "Optional header", "Marked as important");
+/// doc_section!(? "Marked as info");
+/// doc_section!(? "Optional header", "Marked as info");
+/// doc_section!(> "Marked as example");
+/// doc_section!(> "Optional header", "Marked as example");
+/// ```
+#[macro_export]
```
Comment (wdanilo) [inline app/gui/src/model/suggestion_database/mock.rs:370]: The `@` case is not covered

File: app/gui/src/model/suggestion_database/mock.rs [hunk 5]
```diff
@@ -321,6 +322,94 @@ macro_rules! mock_suggestion_database {
...
+macro_rules! doc_section {
+    ($paragraph:expr) => {
+        engine_protocol::language_server::DocSection::Paragraph { body: $paragraph.into() }
```
Comment (wdanilo) [inline app/gui/src/model/suggestion_database/mock.rs]: change `engine_protocol::language_server::DocSection` to `$crate` and re-export it from the crate. Alternatively, change it to `$crate::app::gui::suggestion_database::...`. Macros that are exported should always succeed even if `engine_protocol` was not used in cargo toml config.

File: app/gui/src/model/suggestion_database/mock.rs [hunk 6]
```diff
@@ -321,6 +322,94 @@ macro_rules! mock_suggestion_database {
...
+    };
+    (@ $tag:expr, $body:expr) => {
+        engine_protocol::language_server::DocSection::Tag { name: $tag.into(), body: $body.into() }
```
Comment (wdanilo) [inline app/gui/src/model/suggestion_database/mock.rs:373]: format it the same way as rustfmt pls

File: app/gui/src/model/suggestion_database/mock.rs [hunk 7]
```diff
@@ -321,6 +322,94 @@ macro_rules! mock_suggestion_database {
...
+            header: Some($header.into()),
+            body:   $body.into(),
+        }
+    };
+    (? $body:expr) => {
+        engine_protocol::language_server::DocSection::Marked {
+            mark:   engine_protocol::language_server::Mark::Info,
+            header: None,
+            body:   $body.into(),
+        }
```
Comment (wdanilo) [inline app/gui/src/model/suggestion_database/mock.rs]: A lot of repetitions, please refactor it tto to two macros, where the second macro genwerates body and the first provides it with the `mark` name.

**Review by wdanilo — APPROVED** [review]

File: app/gui/src/documentation_debug_scene.rs (hunk 1 above)
Comment (vitvakatu) [reply app/gui/src/documentation_debug_scene.rs]: Fixed

File: app/gui/src/documentation_debug_scene.rs (hunk 2 above)
Comment (vitvakatu) [reply app/gui/src/documentation_debug_scene.rs]: Fixed

File: app/gui/src/documentation_debug_scene.rs (hunk 3 above)
Comment (vitvakatu) [reply app/gui/src/documentation_debug_scene.rs]: Fixed

File: app/gui/src/model/suggestion_database/mock.rs (hunk 4 above)
Comment (vitvakatu) [reply app/gui/src/model/suggestion_database/mock.rs:370]: It is covered, it is used to create `DocSection::Tag`

File: app/gui/src/model/suggestion_database/mock.rs (hunk 5 above)
Comment (vitvakatu) [reply app/gui/src/model/suggestion_database/mock.rs]: Fixed

File: app/gui/src/model/suggestion_database/mock.rs (hunk 6 above)
Comment (vitvakatu) [reply app/gui/src/model/suggestion_database/mock.rs:373]: This is rustfmt formatting

File: app/gui/src/model/suggestion_database/mock.rs (hunk 7 above)
Comment (vitvakatu) [reply app/gui/src/model/suggestion_database/mock.rs]: Fixed

**Pull request merged**